RELEASING.md

```diff
@@ -24,9 +38,19 @@ In a nutshell:
 
 So, to make a Felix release:
 
+- Consider whether you should update the libcalico-go pin in glide.yaml.
+  If you do so, you should run `make update-vendor` to update the 
+  `glide.lock` file.  Be wary of any additional libraries that get 
+  revved if they aren't being pulled in by the libcalico-go update. At 
+  this late stage, it's safer to only update commit IDs that you're
+  explictly expecting (i.e. undo any changes that `make update-vendor`
+  makes that you weren't expecting).
+
 - Run `make release VERSION=<new version>` and follow the instructions.  This
   creates the annotated release tag, builds the release artifacts, and tells
-  you what else you need to do to publish those.
+  you what else you need to do to publish those.  The release script
+  expects a version number of the form "2.0.0", with optional suffixes
+  such as "-beta.1-rc3".
```

Comment:
If it's OK with you, I'd rather not recommend extra '.'s, so would say `-beta1-rc3` here.